app/models/ems_refresh/save_inventory.rb

```diff
@@ -219,7 +220,10 @@ def save_disks_inventory(hardware, hashes)
       h[:storage_profile_id] = h.fetch_path(:storage_profile, :id)
     end
 
-    save_inventory_multi(hardware.disks, hashes, :use_association, [:controller_type, :location], nil, [:storage, :backing, :storage_profile])
+    filenames = hashes.collect { |h| h[:filename] }
+
+    save_inventory_mulit_child(:disks, Disk.where(:filename => filenames), parent, hashes,
+                               [:filename, :device_name], nil, [:storage, :backing, :storage_profile])
```

Comment:
Yeah this is going to take some thought, I can say for sure this won't work for VMware because `:filename => [GlobalDS_0] DC1_C7_RP1_VM3/DC1_C7_RP1_VM3.vmdk` and `:device_name => Hard disk 1` both of which could be duplicated on another ems.